File: SpringBoot/demo/src/main/java/com/example/rps/beans/buy/BuyTransaction.java
package com.example.rps.beans.buy;

import com.example.rps.beans.Transaction;

public class BuyTransaction extends Transaction 
{
	private int consumerId;
	
	private OnlinePayment onlinePayment;
	
	public int getConsumerId() {
		return consumerId;
	}

	public void setConsumerId(int consumerId) {
		this.consumerId = consumerId;
	}


	public OnlinePayment getOnlinePayment() {
		return onlinePayment;
	}

	public void setOnlinePayment(OnlinePayment onlinePayment) {
		this.onlinePayment = onlinePayment;
	}

}

File: Angular/satva/src/app/buytransaction.ts
import { Onlinepayment } from "./onlinepayment";
import { Transaction } from "./transaction";

export class Buytransaction extends Transaction {
	private consumerId: number = 0;
	public get consumerid(): number {
		return this.consumerId;
	}
	public set consumerid(value: number) {
		this.consumerId = value;
	}
    private onlinePayment: Onlinepayment = new Onlinepayment();
	public get onlinepayment(): Onlinepayment {
		return this.onlinePayment;
	}
	public set onlinepayment(value: Onlinepayment) {
		this.onlinePayment = value;
	}

	
}
